realsense/plugin.cpp

```diff
@@ -35,17 +43,111 @@ class realsense : public plugin {
-        {
+        , realsense_cam{ILLIXR::getenv_or("REALSENSE_CAM", "auto")}
+        {      
             cfg.disable_all_streams();
-            cfg.enable_stream(RS2_STREAM_ACCEL, RS2_FORMAT_MOTION_XYZ32F, ACCEL_RATE); // adjustable to 0, 63 (default), 250 hz
-            cfg.enable_stream(RS2_STREAM_GYRO, RS2_FORMAT_MOTION_XYZ32F, GYRO_RATE); // adjustable set to 0, 200 (default), 400 hz
-            cfg.enable_stream(RS2_STREAM_INFRARED, 1, IMAGE_WIDTH, IMAGE_HEIGHT, RS2_FORMAT_Y8, FPS);
-            cfg.enable_stream(RS2_STREAM_INFRARED, 2, IMAGE_WIDTH, IMAGE_HEIGHT, RS2_FORMAT_Y8, FPS);
-            cfg.enable_stream(RS2_STREAM_COLOR, IMAGE_WIDTH, IMAGE_HEIGHT, RS2_FORMAT_BGR8, FPS);
-            cfg.enable_stream(RS2_STREAM_DEPTH, IMAGE_WIDTH, IMAGE_HEIGHT, RS2_FORMAT_Z16, FPS);
+            configure_camera();
+        }
+    
+    void find_supported_devices(rs2::device_list devices){
+        for (rs2::device device : devices) {
+            if (device.supports(RS2_CAMERA_INFO_PRODUCT_LINE)){
+                std::string product_line = device.get_info(RS2_CAMERA_INFO_PRODUCT_LINE); 
+                #ifndef NDEBUG
+                    std::cout << "Found Product Line: " << product_line << std::endl;
+                #endif
+                if (product_line == "D400"){
+                    #ifndef NDEBUG
+                        std::cout << "Checking for supported streams" << std::endl;
+                    #endif
+                    std::vector<rs2::sensor> sensors = device.query_sensors();
+                    for (rs2::sensor sensor : sensors){
+                        std::vector<rs2::stream_profile> stream_profiles = sensor.get_stream_profiles();
+                        //Currently, all D4XX cameras provide infrared, RGB, and depth, so we only need to check for accel and gyro
+                        for (auto&& sp : stream_profiles)
+                        {
+                            if (sp.stream_type() == RS2_STREAM_GYRO){
+                                gyro_found = true;
+                            };
+                            if (sp.stream_type() == RS2_STREAM_ACCEL){
+                                accel_found = true;
+                            }
+                        }
+                    }
+                    if (accel_found && gyro_found){
+                        D4XXI_found = true;
+                        #ifndef NDEBUG
+                            std::cout << "Supported D4XX found!" << std::endl;
+                        #endif
+                    }
+                } 
+                else if (product_line == "T200"){
+                    T26X_found = true;
+                    #ifndef NDEBUG
+                        std::cout << "T26X found! " << std::endl;
+                    #endif
+                }
+            }
+            
+        }
+        if (!T26X_found && !D4XXI_found){
+            std::cout << "No supported Realsense device detected!" << std::endl;
+        }
```

Comment:
Only required during debug.

Reply:
I think this print should be there always because it signifies a fatal error.

Reply:
That was my initial reaction too, but this function only finds devices. `configure_camera` does the error handling, and aborts on line 130 if no device is found. I think it's a clean solution, but open to suggestions.